utility/src/main/java/org/oppia/android/util/parser/BlurTransformation.kt

```diff
@@ -10,13 +10,11 @@ import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
 import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
 import java.security.MessageDigest
 
-class BlurTransformation(context: Context) : BitmapTransformation() {
-  private val rs: RenderScript
+/** [BlurTransformation] is a bitmap transformation that blurs an image using RenderScript. */
+class BlurTransformation(
```

Comment:
I think there are some other changes we can do to make this more Oppia Android-idiomatic:

```kotlin
// TODO: add documentation.
private const BLUR_RADIUS = 20f;

/** [BlurTransformation] is a bitmap transformation that blurs an image using RenderScript. */
// TODO: please add any references that we've used to the above.
class BlurTransformation @Inject constructor(
  private val context: Context
) : BitmapTransformation() {
  private val renderScript by lazy { RenderScript.create(context) }
  
  override fun transform(
    pool: BitmapPool,
    toTransform: Bitmap,
    outWidth: Int,
    outHeight: Int
  ): Bitmap {
    // TODO: ensure that Bitmap config is ARGB_8888, otherwise we may need to use something other than Element.U8_4 below OR do a copy like before to convert.

    // Create a new RenderScript allocation pointing to the original bitmap to avoid a memory copy.
    val inputAllocation = Allocation.createFromBitmap(renderScript, toTransform)

    // Create a new RenderScript allocation to receive the output from the blur operation.
    val outputAllocation = Allocation.createTyped(renderScript, inputAllocation.type)

    // Create a new Gaussian blur script with 4 expect 8-bit color channels (e.g. ARGB).
    val blurScript = ScriptIntrinsicBlur.create(renderScript, Element.U8_4(renderScript))
    blurScript.setInput(input)
    blurScript.setRadius(BLUR_RADIUS)
    blurScript.forEach(output)

    // Transform the original bitmap to avoid needing to reallocate a second one.
    output.copyTo(toTransform)
    return toTransform
  }

  override fun updateDiskCacheKey(messageDigest: MessageDigest) {
    messageDigest.update("blur transformation".toByteArray())
  }
}
```

Note that I heavily used https://stackoverflow.com/a/23119957 as a reference here. I also added TODOs where some additional details need to be filled in. We need to double check that this implementation works as expected for existing use cases since it's been optimized to avoid allocating an extra bitmap.

Reply:
Thank you, I applied your suggestions and added a condition ensure that Bitmap config is ARGB_8888. But I'm not entirely sure that this is what you meant in your TODO.

Reply:
Your solution looks good--thanks!